examples/advanced/main.go

```diff
@@ -36,19 +38,44 @@ import (
 // main is compiled to an exported Wasm function named "_start", called by the
 // Wasm scheduler plugin during initialization.
 func main() {
+	// Instead of using `plugin.Set`, this configures only the interfaces
+	// implemented by the plugin. The Wasm host only calls functions imported,
+	// so this prevents additional overhead.
+	enqueue.SetPlugin(func(h handleapi.Handle) api.EnqueueExtensions {
+		p := pluginInitializer(h)
+		plugin, ok := p.(api.EnqueueExtensions)
+		if ok {
+			return plugin
+		}
+		panic("pluginInitializer did not return an api.EnqueueExtensions")
```

Comment:
```suggestion
		return p.(api.EnqueueExtensions)
```

It panics anyway if p doesn't implement EnqueueExtensions.